setup.py

```diff
@@ -5,29 +5,55 @@
 
 from setuptools import find_packages, setup
 from setuptools.command.build_ext import build_ext as build_ext_orig
-from setuptools.extension import Extension
+from setuptools.command.install_lib import install_lib as install_lib_orig
 
 import versioneer
 
 
+def install_java_libraries(dir):
+    """Helper function to run dask-sql's java installation process in a given directory"""
+
+    # build the jar
+    maven_command = shutil.which("mvn")
+    if not maven_command:
+        raise OSError(
+            "Can not find the mvn (maven) binary. Make sure to install maven before building the jar."
+        )
+    command = [maven_command, "clean", "install", "-f", "pom.xml"]
+    subprocess.check_call(command, cwd=os.path.join(dir, "planner"))
+
+    # copy generated jar to python package
+    os.makedirs(os.path.join(dir, "dask_sql/jar"), exist_ok=True)
+    shutil.copy(
+        os.path.join(dir, "planner/target/DaskSQL.jar"),
+        os.path.join(dir, "dask_sql/jar/"),
+    )
+
+
 class build_ext(build_ext_orig):
-    """Build the external Java libraries"""
+    """Build and install the java libraries for an editable install"""
 
     def run(self):
-        """Run the mvn installation command"""
-        maven_command = shutil.which("mvn")
-        if not maven_command:
-            raise OSError(
-                "Can not find the mvn (maven) binary. Make sure to install maven before building the jar."
-            )
-        command = [maven_command, "clean", "install", "-f", "pom.xml"]
+        super().run()
+
+        # build java inplace
+        install_java_libraries("")
+
+
+class install_lib(install_lib_orig):
+    """Build and install the java libraries for a standard install"""
+
+    def build(self):
+        super().build()
 
-        subprocess.check_call(command, cwd="planner")
+        # copy java source to build directory
+        self.copy_tree("planner", os.path.join(self.build_dir, "planner"))
 
-        os.makedirs("dask_sql/jar", exist_ok=True)
-        shutil.copy("planner/target/DaskSQL.jar", "dask_sql/jar/DaskSQL.jar")
+        # build java in build directory
+        install_java_libraries(self.build_dir)
```

Comment:
Does this result in building the JAR twice given `mvn` is called in this function?

Reply:
Now that I've removed the empty extension, `build_ext` should only be called on editable installs where `install_lib` is not called - so I think we should be safe 🙂 

Can verify that running both

```
pip install -e . --no-deps -vv
pip install . --no-deps -vv
```

I only see the jar get built once - are there any other commands we should look out for?